plugins/device/cmd/nvidia/device.go

```diff
@@ -0,0 +1,206 @@
+
+	log "github.com/hashicorp/go-hclog"
+
+	"github.com/hashicorp/nomad/plugins/base"
+	"github.com/hashicorp/nomad/plugins/device"
+	"github.com/hashicorp/nomad/plugins/device/cmd/nvidia/nvml_client"
+	"github.com/hashicorp/nomad/plugins/shared/hclspec"
+)
+
+const (
+	// pluginName is the name of the plugin
+	pluginName = "nvidia-gpu"
+
+	// vendor is the vendor providing the devices
+	vendor = "nvidia"
+
+	// deviceType is the type of device being returned
+	deviceType = device.DeviceTypeGPU
+
+	// notAvailable value is returned to nomad server in case some properties were
+	// undetected by nvml driver
+	notAvailable = "N/A"
+)
+
+const (
+	// Nvidia-contaner-runtime environment variable names
+	nvidiaVisibleDevices = "NVIDIA_VISIBLE_DEVICES"
+)
+
+var (
+	// pluginInfo describes the plugin
+	pluginInfo = &base.PluginInfoResponse{
+		Type:             base.PluginTypeDevice,
+		PluginApiVersion: "0.0.1", // XXX This should be an array and should be consts
+		PluginVersion:    "0.1.0",
+		Name:             pluginName,
+	}
+
+	// configSpec is the specification of the plugin's configuration
+	configSpec = hclspec.NewObject(map[string]*hclspec.Spec{
+		"ignored_gpu_ids": hclspec.NewDefault(
+			hclspec.NewAttr("ignored_gpu_ids", "list(string)", false),
+			hclspec.NewLiteral("[]"),
+		),
+		"fingerprint_period": hclspec.NewDefault(
+			hclspec.NewAttr("fingerprint_period", "string", false),
+			hclspec.NewLiteral("\"5s\""),
+		),
+		"stats_period": hclspec.NewDefault(
+			hclspec.NewAttr("stats_period", "string", false),
+			hclspec.NewLiteral("\"5s\""),
+		),
+	})
+)
+
+// Config contains configuration information for the plugin.
+type Config struct {
+	IgnoredGPUIDs     []string `codec:"ignored_gpu_ids"`
+	FingerprintPeriod string   `codec:"fingerprint_period"`
+	StatsPeriod       string   `codec:"stats_period"`
+}
+
+// NvidiaDevice contains all plugin specific data
+type NvidiaDevice struct {
+	// nvmlClient is used to get data from nvidia
+	nvmlClient nvml.NVMLClientInterface
+
+	// nvmlClientInitializationError holds an error retrieved during
+	// nvmlClient initialization
+	nvmlClientInitializationError error
+
+	// ignoredGPUIDs is a list of UUIDs that would not be exposed to nomad
+	ignoredGPUIDs []string
+
+	// fingerprintPeriod is how often we should call nvml to get list of devices
+	fingerprintPeriod time.Duration
+
+	// statsPeriod is how often we should collect statistics for fingerprinted
+	// devices.
+	statsPeriod time.Duration
+
+	// devices is the set of detected eligible devices
+	devices    map[string]struct{}
+	deviceLock sync.RWMutex
+
+	logger log.Logger
+}
+
+// NewNvidiaDevice returns a new nvidia device plugin.
+func NewNvidiaDevice(log log.Logger) *NvidiaDevice {
+	nvmlClient, nvmlClientInitializationError := nvml.NewInitializedNvmlClient()
+	logger := log.Named(pluginName)
+	if nvmlClientInitializationError != nil {
+		logger.Error("Unable to initialize Nvidia driver with error", nvmlClientInitializationError)
```

Comment:
All logs should start lowercase. The logging library is key value, so it should start with the message and then pair of arguments.
`logger.Error("unable to initialize Nvidia driver", "error", nvmlClientInitializationError)`